src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_PatternSwitchStatement.cs

```diff
@@ -300,7 +308,18 @@ private void LowerDecisionTree(DecisionTree.Guarded guarded)
                     _loweredDecisionTree.Add(_factory.Goto(checkGuard));
                     sectionBuilder.Add(_factory.Label(checkGuard));
                     AddBindings(sectionBuilder, guarded.Bindings);
-                    sectionBuilder.Add(_factory.ConditionalGoto(LocalRewriter.VisitExpression(guarded.Guard), targetLabel, true));
+
+                    var guardTest = _factory.ConditionalGoto(LocalRewriter.VisitExpression(guarded.Guard), targetLabel, true);
+
+                    // Create the sequence point if generating debug info and
+                    // node is not compiler generated
```

Comment:
Minor point: instrumenting is not limited to sequence points. Perhaps change the comment to `Instrument if generating debug info ...`.